src/App.tsx

```diff
@@ -1,24 +1,289 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useMemo,
+  useRef,
+  useContext,
+} from 'react';
+import { Todo } from './types/Todo';
+import { Notification } from './components/Notification';
+import { TodoList } from './components/TodoList';
 import { UserWarning } from './UserWarning';
+import {
+  deleteTodoById,
+  addNewTodo,
+  updateTodoById,
+  getTodos,
+} from './api/todos';
+import { Loader } from './components/Loader';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { LoadingTodosContext, TodosContext } from './context/TodosContexts';
 
-const USER_ID = 0;
+const USER_ID = 11497;
+
+function getUpdatedTodos(
+  todos: Todo[],
+  updatedTodo: Todo,
+  updatedField: keyof Todo,
+) {
+  return todos.map(todo => {
+    if (todo.id === updatedTodo.id) {
+      return {
+        ...todo,
+        [updatedField]: updatedTodo[updatedField],
+      };
+    }
+
+    return todo;
+  });
+}
 
 export const App: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const {
+    todosIdToDelete,
+    setTodosIdToDelete,
+    setTodosIdToUpdate,
+  } = useContext(LoadingTodosContext);
```

Comment:
You can create hooks for contexts so you don't have to pass every time the context.
```javascript
const useTodos = () => useContext(TodosContext);
const useLoadingTodos = () => useContext(LoadingTodosContext);
```